commands/compile/compile.go

```diff
@@ -128,7 +128,12 @@ func Compile(ctx context.Context, req *rpc.CompileRequest, outStream, errStream
 	// FIXME: This will be redundant when arduino-builder will be part of the cli
 	builderCtx.OtherLibrariesDirs = paths.NewPathList(req.GetLibraries()...)
 	builderCtx.OtherLibrariesDirs.Add(configuration.LibrariesDir(configuration.Settings))
-	builderCtx.LibraryDirs = paths.NewPathList(req.Library...)
+	for _, libPath := range paths.NewPathList(req.Library...) {
+		if libPath, err = libPath.Abs(); err != nil {
+			return nil, &arduino.LibraryNotFoundError{Cause: err}
+		}
+		builderCtx.LibraryDirs = append(builderCtx.LibraryDirs, libPath)
+	}
```

Comment:
Move this change from `commands` package (gRPC), to `internal/cli` package (arduino-cli).

It doesn't make much sense to use relative paths in a gRPC call as a daemon. It makes sense in the `arduino-cli` as a command line, so the paths must be absolutized before entering the gRPC request.